core/common/src/main/java/alluxio/network/protocol/databuffer/DataBuffer.java

```diff
@@ -40,6 +40,22 @@
   ByteBuffer getReadOnlyByteBuffer();
 
   /**
+   * Transfers this buffer's data to the specified destination starting at
+   * the current {@code readerIndex} and increases the {@code readerIndex}
+   * by the number of the transferred bytes (= {@code length}).
+   *
+   * @param dst the destination
+   * @param dstIndex the first index of the destination
+   * @param length the number of bytes to transfer
+   */
+  void readBytes(byte[] dst, int dstIndex, int length);
+
+  /**
+   * @return the number of readable bytes remaining
+   */
+  int readableBytes();
```

Comment:
But Java int max is 2,147,483,647 which is 2GB.
ByteBuffer and ByteBuf may have legacy reasons to return int rather than long. But this interface is an abstraction and I don't see why we should be bound by the two current existing subclass implementation? Typically, for length in bytes, we should always use long rather than int.